src/emqx_cm.erl

```diff
@@ -286,44 +295,63 @@ discard_session(ClientId) when is_binary(ClientId) ->
         ChanPids -> lists:foreach(fun(Pid) -> discard_session(ClientId, Pid) end, ChanPids)
     end.
 
-%% @private Kick a local stale session to force it step down.
-%% If failed to kick (e.g. timeout) force a kill.
+%% @private call a local stale session to execute an Action.
+%% If failed to response (e.g. timeout) force a kill.
 %% Keeping the stale pid around, or returning error or raise an exception
 %% benefits nobody.
--spec kick_or_kill(kick | discard, module(), pid()) -> ok.
-kick_or_kill(Action, ConnMod, Pid) ->
+-spec call_or_kill(Action, module(), pid())
+    -> ok                        %% returned by kick, discard
+     | emqx_session:session()    %% returned by {takeover, 'begin'}
+     | list(emqx_type:deliver()) %% returned by {takeover, 'end'}
+  when Action :: kick | discard | {takeover, 'begin'} | {takeover, 'end'}.
+call_or_kill(Action, ConnMod, Pid) ->
```

Comment:
Maybe just name it ‘takeover’

Kick, kill are actually a part of the takeover process.
The mfa reported in the logs would be more clear tooo